app/js/components/orderEntry/OrderEntryPage.jsx

```diff
@@ -58,10 +60,20 @@ export class OrderEntryPage extends PureComponent {
     } = this.props.createOrderReducer;
 
     const { intl } = this.props;
+    const { afterAddOrderUrl } = this.state;
     const orderCreatedMsg = intl.formatMessage({ id: "app.orders.create.success", defaultMessage: "Order Successfully Created" });
     if (added && labOrderData !== prevProps.createOrderReducer.labOrderData) {
       successToast(orderCreatedMsg);
-      this.props.fetchLabOrders(null, this.props.patient.uuid);
+      // if an afterAddOrderUrl has been specified, redirect to that page
+      if (afterAddOrderUrl && labOrderData && labOrderData.orders) {
+        // we pass in the uuid of the new order (just picking the "first" order if multiple were created)
+        let url = Handlebars.compile(afterAddOrderUrl)({
+          order: labOrderData.orders[0].uuid,
```

Comment:
Ok as long as you're confident that `labOrderData.orders` cannot be an empty array, and that if it has a member, that member must have a `uuid` element.

Reply:
I think so, but, fair enough, I'll add a check.

Reply:
Actually, it really *must* have a uuid, but good to check that is has a member.